### PR TITLE
Bug 1921781: i18n DefaultList

### PR DESCRIPTION
**Pull request by rebeccaalpert**
The DefaultList component was not internationalized. This impacted the Search page results.

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1921781.

### CODE REVIEW

File: frontend/public/components/default-resource.jsx [hunk 1]
```diff
@@ -25,69 +27,6 @@ const tableColumnClasses = [
...
-        <Timestamp timestamp={obj.metadata.creationTimestamp} />
-      </TableData>
-      <TableData className={tableColumnClasses[3]}>
-        <ResourceKebab actions={menuActions} kind={kind} resource={obj} />
-      </TableData>
-    </TableRow>
-  );
-};
-
 export const DetailsForKind = (kind) =>
```
Comment (spadgett) [inline frontend/public/components/default-resource.jsx:30]: It looks like we missed the text in the `SectionHeading` just below, and again for "Conditions."

File: frontend/public/components/default-resource.jsx [hunk 2]
```diff
@@ -117,10 +56,74 @@ export const DetailsForKind = (kind) =>
...
+        </TableData>
+      </TableRow>
+    );
+  };
+
   return (
     <Table
       {...props}
-      aria-label="Default Resource"
+      aria-label={t('public~Default Resource')}
```
Comment (spadgett) [inline frontend/public/components/default-resource.jsx:126]: Not new, but the label should really be the actual kind instead of "Default Resource"

**Comment by spadgett** [comment]
/override ci/prow/ceph-storage-plugin

The ceph-storage-plugin job should only have run on PRs with changes to `frontend/packages/ceph-storage-plugin`. It failed due to an unrelated CI config change. Overriding status on this PR.

File: frontend/public/components/default-resource.jsx [hunk 3]
```diff
@@ -25,75 +27,25 @@ const tableColumnClasses = [
...
+      }
+      return t(model.labelKey);
+    };
+
     return (
       <>
         <div className="co-m-pane__body">
-          <SectionHeading text={`${kindForReference(kind)} Details`} />
+          <SectionHeading
+            text={t('public~{{kindLabel}} Details', { kindLabel: getKindLabel(kind) })}
```
Comment (spadgett) [inline frontend/public/components/default-resource.jsx:47]:
```suggestion
            text={t('public~{{kindLabel}} details', { kindLabel: getKindLabel(kind) })}
```

File: frontend/public/components/default-resource.jsx [hunk 4]
```diff
@@ -25,75 +27,25 @@ const tableColumnClasses = [
...
   function DetailsForKind_({ obj }) {
+    const { t } = useTranslation();
+
+    const getKindLabel = (item) => {
+      const model = modelFor(item);
+      // API discovery happens asynchronously. Avoid runtime errors if the model hasn't loaded.
+      if (!model || !model.labelKey) {
+        return kindForReference(item);
+      }
+      return t(model.labelKey);
```
Comment (spadgett) [inline frontend/public/components/default-resource.jsx:40]:
```suggestion
      return model?.labelKey || kindForReference(item);
```

**Comment by rebeccaalpert** [comment]
Comments all set!

**Comment by rebeccaalpert** [comment]
Seems like a testing infrastructure failure rather than a test failure.

/test e2e-gcp-console

File: frontend/public/components/default-resource.jsx [hunk 5]
```diff
@@ -25,75 +27,21 @@ const tableColumnClasses = [
...
-  );
-};
-
 export const DetailsForKind = (kind) =>
   function DetailsForKind_({ obj }) {
+    const { t } = useTranslation();
+
+    const getKindLabel = (item) => {
+      const model = modelFor(item);
+      return model?.labelKey || kindForReference(item);
```
Comment (spadgett) [inline frontend/public/components/default-resource.jsx:36]: Sorry, my code suggestion was wrong.

```suggestion
      return t(model?.labelKey) || kindForReference(item);
```

Comment (rebeccaalpert) [reply frontend/public/components/default-resource.jsx:36]: I should have noticed. :) Fixed.

**Review by spadgett — APPROVED** [review]
/lgtm

File: frontend/public/components/default-resource.jsx [hunk 6]
```diff
@@ -25,75 +27,21 @@ const tableColumnClasses = [
...
-  );
-};
-
 export const DetailsForKind = (kind) =>
   function DetailsForKind_({ obj }) {
+    const { t } = useTranslation();
+
+    const getKindLabel = (item) => {
+      const model = modelFor(item);
+      return t(model?.labelKey) || kindForReference(item);
```
Comment (spadgett) [inline frontend/public/components/default-resource.jsx:36]: Ugh, sorry. I got this wrong twice. My fault.

```suggestion
      return model?.labelKey ? t(model.labelKey) : kindForReference(item);
```

Comment (rebeccaalpert) [reply frontend/public/components/default-resource.jsx:36]: All set @spadgett.

**Review by spadgett — APPROVED** [review]
/lgtm

**Comment by spadgett** [comment]
@rebeccaalpert Will need rebase, thanks!

**Comment by rebeccaalpert** [comment]
It looks like a partial fix was committed yesterday. Unfortunately it involves a lot of passing the t function around. I'm refactoring my work back in and trying to clean that up now. 

See https://github.com/openshift/console/commit/7670b61adf96e15a871413874a559b566dc44457#diff-2d955fdd8c9fde58ccfb693e3bee7cf3f5b7e29da8e865da19a51cbe8d1539b0.

**Comment by rebeccaalpert** [comment]
Refactored code and reset some of the work that passed TFunction around since it's not a best practice.

File: frontend/public/components/default-resource.tsx [hunk 7]
```diff
@@ -35,120 +35,131 @@ const tableColumnClasses = [
...
-  );
-};
+export const DetailsForKind = (kind: string) =>
+  function DetailsForKind_({ obj }) {
+    const { t } = useTranslation();
 
-type DetailsForKindProps = {
-  obj: K8sResourceKind;
-};
+    const getKindLabel = (item) => {
```
Comment (spadgett) [inline frontend/public/components/default-resource.tsx:42]:
```suggestion
    const getKindLabel = (item: K8sResourceKindReference) => {
```

**Review by spadgett — APPROVED** [review]
/lgtm

**Comment by spadgett** [comment]
/lgtm cancel

i18n diff

**Comment by rebeccaalpert** [comment]
Sorry, my bad. Added the type and updated the i18n files.

**Comment by rebeccaalpert** [comment]
Will debug test failure tomorrow.

**Review by spadgett — APPROVED** [review]
/lgtm

**Pull request merged**